executor/distsql.go

```diff
@@ -174,27 +174,37 @@ func splitRanges(ranges []*ranger.Range, keepOrder bool) ([]*ranger.Range, []*ra
 		if !keepOrder {
 			return append(unsignedRanges, signedRanges...), nil
 		}
+		if desc {
+			return unsignedRanges, signedRanges
+		}
 		return signedRanges, unsignedRanges
 	}
 	signedRanges := make([]*ranger.Range, 0, idx+1)
 	unsignedRanges := make([]*ranger.Range, 0, len(ranges)-idx)
 	signedRanges = append(signedRanges, ranges[0:idx]...)
-	signedRanges = append(signedRanges, &ranger.Range{
-		LowVal:     ranges[idx].LowVal,
-		LowExclude: ranges[idx].LowExclude,
-		HighVal:    []types.Datum{types.NewUintDatum(math.MaxInt64)},
-	})
-	unsignedRanges = append(unsignedRanges, &ranger.Range{
-		LowVal:      []types.Datum{types.NewUintDatum(math.MaxInt64 + 1)},
-		HighVal:     ranges[idx].HighVal,
-		HighExclude: ranges[idx].HighExclude,
-	})
+	if !(ranges[idx].LowVal[0].GetUint64() == math.MaxInt64 && ranges[idx].LowExclude) {
```

Comment:
Why do we need this check?

Reply:
Because there can be cases like `(maxInt64, maxInt64]` and [maxInt64+1, maxInt64+1)`. And then error thrown when send requests.